### PR TITLE
feature(sct-runner): update image to use java-21

### PR DESCRIPTION
**Pull request by fruch**
we are switching all jenkins workers to start using java-21 while we are at it, we are updates the sct-runners images to be based on ubuntu 2404 images

### Testing

- [x] build image aws
- [x] build image gce
- [x] build image azure
- [x] provision tests
- [x] test on aws (worker + runner): 🟢 : https://jenkins.scylladb.com/job/scylla-staging/job/fruch/job/longevity-10gb-3h-ipv6-test/18/
- [x] test on gce (worker + runner): 🟢 : https://jenkins.scylladb.com/job/scylla-staging/job/fruch/job/longevity-10gb-3h-gce-test/45/
- [x] test on azure (AWS worker + runner): 🟢 https://jenkins.scylladb.com/job/scylla-staging/job/fruch/job/longevity-10gb-3h-azure-test/33/
- [x] 🟢  https://jenkins.scylladb.com/view/staging/job/scylla-staging/job/fruch/job/artifacts-azure-image-test/49/

### PR pre-checks (self review)

- [x] I added the relevant `backport` labels
- [x] I didn't leave commented-out/debugging code

### Reminders

- Add New configuration option and document them (in `sdcm/sct_config.py`)
- Add unit tests to cover my changes (under `unit-test/` folder)
- Update the Readme/doc folder relevant to this change (if needed)

### CODE REVIEW

File: docs/sct-runners.md [hunk 1]
```diff
@@ -22,3 +22,25 @@ hydra --execute-on-runner `cat sct_runner_ip` [Any hydra commands]
...
+    SCT_GCE_PROJECT=gcp-local-ssd-latency ./sct.py create-runner-image -c gce -r us-east1 -z a
+
+    ./sct.py create-runner-image -c azure -r eastus -z a
+    ```
+3. update version on `aws_builder.py` and `gce_builder.py`
+4. build jenkins configuration with new sct runner image
+    ```bash
+    ./sct.py configure-jenkins-builders -c gce
+    ./sct.py configure-jenkins-builders -c aws
+    ```
```
Comment (soyacz) [inline docs/sct-runners.md:46]: one more thing to update labels in jenkinsfiles and `vars/getJenkinsLabels.groovy`

File: sdcm/sct_runner.py [hunk 2]
```diff
@@ -221,6 +221,7 @@ def install_prereqs(self, public_ip: str, connect_timeout: Optional[int] = None)
              # Disable apt-key warnings and set non-interactive frontend.
             export APT_KEY_DONT_WARN_ON_DANGEROUS_USAGE=1
             export DEBIAN_FRONTEND=noninteractive
+            export PIP_BREAK_SYSTEM_PACKAGES=true
```
Comment (soyacz) [inline sdcm/sct_runner.py:224]: I wonder why it started to be needed

**Review by soyacz — APPROVED** [review]
looks like safer way than updating existing builders 😅

LGTM

File: sdcm/sct_runner.py (hunk 2 above)
Comment (fruch) [reply sdcm/sct_runner.py:224]: it's python 3.12, which is the default in ubuntu 2404

**Comment by fruch** [comment]
> looks like safer way than updating existing builders 😅
> 
> LGTM

it was planned for that, but it's the first time we actually use it in such safe way (after finding the hard way it wasn't safe enoguh)

File: docs/sct-runners.md (hunk 1 above)
Comment (fruch) [reply docs/sct-runners.md:46]: yeah, that's also part of game

**Review by vponomaryov — APPROVED** [review]
LGTM

**Pull request merged**